validator/sawtooth_validator/consensus/proxy.py

```diff
@@ -29,6 +33,65 @@ class UnknownBlock(Exception):
     """The given block could not be found."""
 
 
+class ConsensusActivationObserver(ChainObserver):
+    """Observes chain updates and activates new engines, if they have been activated."""
+
+    def __init__(self, consensus_registry, consensus_notifier, settings_view_factory):
+        self._registry = consensus_registry
+        self._notifier = consensus_notifier
+        self._settings_view_factory = settings_view_factory
+
+    def chain_update(self, block, receipts):
+        # Check to see if there is an active engine at all:
+        if not self._registry.has_active_engine():
+            self._update_active_engine(block)
+            return
+
+        # Otherwise, check to see fi the settings changed, before updating
+        # the active engine.
+        receipt_events = ReceiptEventExtractor(receipts).extract([
+            EventSubscription(event_type="setting/update")])
+
+        consensus_settings_changed = False
```

Comment:
This bit can be simplified: `for ...: if self._is_consensus_algorithm_setting(event): self._update_active_engine(block); break`

Reply:
Fixed